From b4b3133c48345a6150975d9446830b53b2ff3cfa Mon Sep 17 00:00:00 2001
From: Stefano Babic <stefano.babic@swupdate.org>
Date: Sun, 24 Dec 2023 21:52:28 +0100
Subject: [PATCH 07/15] Backport libubootenv 0.3.5

SWUpdate 2023.12 requires libubootenv 0.3.5. To support last version of
SWUpdate, libubootenv must be ported to the releases with older version
of the library.

Signed-off-by: Stefano Babic <stefano.babic@swupdate.org>
Reported-by: Ayoub Zaki <ayoub.zaki@embexus.com>

Upstream-Status: Backported from Scarthgap to Mickledore
---
 recipes-bsp/u-boot/libubootenv_0.3.5.bb | 26 +++++++++++++++++++++++++
 1 file changed, 26 insertions(+)
 create mode 100644 recipes-bsp/u-boot/libubootenv_0.3.5.bb

diff --git a/recipes-bsp/u-boot/libubootenv_0.3.5.bb b/recipes-bsp/u-boot/libubootenv_0.3.5.bb
new file mode 100644
index 0000000..b68bbb4
--- /dev/null
+++ b/recipes-bsp/u-boot/libubootenv_0.3.5.bb
@@ -0,0 +1,26 @@
+SUMMARY = "U-Boot libraries and tools to access environment"
+
+DESCRIPTION = "This package contains tools and libraries to read \
+and modify U-Boot environment. \
+It provides a hardware-independent replacement for fw_printenv/setenv utilities \
+provided by U-Boot"
+
+HOMEPAGE = "https://github.com/sbabic/libubootenv"
+LICENSE = "LGPL-2.1-or-later"
+LIC_FILES_CHKSUM = "file://LICENSES/LGPL-2.1-or-later.txt;md5=4fbd65380cdd255951079008b364516c"
+SECTION = "libs"
+
+SRC_URI = "git://github.com/sbabic/libubootenv;protocol=https;branch=master"
+SRCREV = "3f4d15e36ceb58085b08dd13f3f2788e9299877b"
+
+S = "${WORKDIR}/git"
+
+inherit cmake lib_package
+
+EXTRA_OECMAKE = "-DCMAKE_BUILD_TYPE=Release"
+
+DEPENDS = "zlib libyaml"
+PROVIDES += "u-boot-fw-utils"
+RPROVIDES:${PN}-bin += "u-boot-fw-utils"
+
+BBCLASSEXTEND = "native"
-- 
2.45.1

